engine/table/src/main/java/io/deephaven/engine/table/impl/SourcePartitionedTable.java

```diff
@@ -187,19 +184,42 @@ private RowSet sortAndAddLocations(@NotNull final Stream<TableLocation> location
             });
             return initialLastRowKey == lastInsertedRowKey.longValue()
                     ? RowSetFactory.empty()
-                    : RowSetFactory.fromRange(initialLastRowKey, lastInsertedRowKey.longValue());
+                    : RowSetFactory.fromRange(initialLastRowKey + 1, lastInsertedRowKey.longValue());
         }
 
         private Table makeConstituentTable(@NotNull final TableLocation tableLocation) {
-            return applyTablePermissions.apply(new PartitionAwareSourceTable(
+            final PartitionAwareSourceTable constituent = new PartitionAwareSourceTable(
                     constituentDefinition,
                     "SingleLocationSourceTable-" + tableLocation,
                     RegionedTableComponentFactoryImpl.INSTANCE,
                     new SingleTableLocationProvider(tableLocation),
-                    refreshSizes ? refreshCombiner : null));
+                    refreshSizes ? refreshCombiner : null);
```

Comment:
I *think* we could stop using a `refreshCombiner` for the constituents if we added installed a `ConstituentDependency` on the result table. Might be more efficient.

Reply:
I'm going to skip this for now -- I am worried that i don't understand it well enough to not make a surprising mistake.